packages/driver/src/dom/elements/find.ts

```diff
@@ -224,55 +223,91 @@ export const getElements = ($el) => {
   return els
 }
 
-export const getContainsSelector = (text, filter = '', options: {
-  matchCase?: boolean
-} = {}) => {
-  const $expr = $.expr[':']
+/*
+ * We add three custom expressions to jquery. These let us use custom
+ * logic around case sensitivity, and match strings or regular expressions.
+ * See https://github.com/jquery/sizzle/wiki/#-pseudo-selectors for
+ * documentation on adding Sizzle selectors.
+ *
+ * Our use of
+ *
+ *   $.expr[':']['cy-contains'] = $.expr.createPseudo()
+ *
+ * is equivelent to
+ *
+ *   Sizzle.selectors.pseudos['cy-contains'] = Sizzle.selectors.createPseudo()
+ *
+ * in the documentation linked above. $.expr[':'] is jquery's alias for
+ * Sizzle.selectors.psuedos.
+ *
+ * These custom expressions are used exclusively by cy.contains; see
+ * `getContainsSelector` below.
+ */
 
-  const escapedText = escapeQuotes(
-    escapeBackslashes(text),
-  )
+// Example:
+// button:cy-contains("Login")
+$.expr[':']['cy-contains'] = $.expr.createPseudo((text) => {
+  text = JSON.parse(`"${ text }"`)
 
-  // they may have written the filter as
-  // comma separated dom els, so we want to search all
-  // https://github.com/cypress-io/cypress/issues/2407
-  const filters = filter.trim().split(',')
+  return function (elem) {
+    let testText = normalizeWhitespaces(elem)
 
-  let cyContainsSelector
+    return testText.includes(text)
+  }
+})
 
-  if (_.isRegExp(text)) {
-    if (options.matchCase === false && !text.flags.includes('i')) {
-      text = new RegExp(text.source, text.flags + 'i') // eslint-disable-line prefer-template
-    }
+// Example:
+// .login-button:cy-contains-insensitive("login")
+$.expr[':']['cy-contains-insensitive'] = $.expr.createPseudo((text) => {
+  text = JSON.parse(`"${ text }"`)
 
-    // taken from jquery's normal contains method
-    cyContainsSelector = function (elem) {
-      if (elem.type === 'submit' && elem.tagName === 'INPUT') {
-        return text.test(elem.value)
-      }
+  return function (elem) {
+    let testText = normalizeWhitespaces(elem)
 
-      const testText = normalizeWhitespaces(elem)
+    testText = testText.toLowerCase()
+    text = text.toLowerCase()
 
-      return text.test(testText)
-    }
-  } else if (_.isString(text)) {
-    cyContainsSelector = function (elem) {
-      let testText = normalizeWhitespaces(elem)
+    return testText.includes(text)
+  }
+})
+
+function isSubmit (elem: Element): elem is HTMLInputElement {
+  return elem.tagName === 'INPUT' && (elem as HTMLInputElement).type === 'submit'
+}
```

Comment:
Why do we have special logic for `isSubmit`?

Also won't any button in a form trigger a submit? eg

```html
<form>
  <button>submit</button>
</form>
```

In this case the button would behave like a submit. 

It is only used once:

```js
// taken from jquery's normal contains method
  return function (elem) {
    if (isSubmit(elem)) {
      return regex.test(elem.value)
    }
```

It's not really clear why we do this here, do you happen to know why? Not sure you needed to dive into that logic while you were here, but I'm curious either way :thinking: